From 1990fb757f6d275d807fcb48ad09f5fc7c947bc6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Wed, 14 Aug 2019 15:57:42 +0200
Subject: [PATCH] udev: use bfq as the default scheduler

NOTE change for azurelinux:

This patch from Fedora has been renamed from "bfq" to "none" and adjusted
to set the udev rule's i/o scheduler from "bfq" to "none" which is the
preferred default i/o scheduler in Azure VMs and for NVMe drives.

Original Fedora commit message below:

As requested in https://bugzilla.redhat.com/show_bug.cgi?id=1738828.
Test results are that bfq seems to behave better and more consistently on
typical hardware. The kernel does not have a configuration option to set
the default scheduler, and it currently needs to be set by userspace.

See the bug for more discussion and links.
---
 rules.d/60-block-scheduler.rules | 5 +++++
 rules.d/meson.build              | 1 +
 2 files changed, 6 insertions(+)
 create mode 100644 rules.d/60-block-scheduler.rules

diff --git a/rules.d/60-block-scheduler.rules b/rules.d/60-block-scheduler.rules
new file mode 100644
index 0000000000..850b64540e
--- /dev/null
+++ b/rules.d/60-block-scheduler.rules
@@ -0,0 +1,5 @@
+# do not edit this file, it will be overwritten on update
+
+ACTION=="add", SUBSYSTEM=="block", ENV{DEVTYPE}=="disk", \
+  KERNEL=="mmcblk*[0-9]|msblk*[0-9]|mspblk*[0-9]|sd*[!0-9]|sr*", \
+  ATTR{queue/scheduler}="none"
diff --git a/rules.d/meson.build b/rules.d/meson.build
index 20fca222da..94fee9d7c0 100644
--- a/rules.d/meson.build
+++ b/rules.d/meson.build
@@ -7,6 +7,7 @@ install_data(
 rules = [
         [files('60-autosuspend.rules',
                '60-block.rules',
+               '60-block-scheduler.rules',
                '60-cdrom_id.rules',
                '60-dmi-id.rules',
                '60-drm.rules',
-- 
2.41.0

